kong/cache/marshall.lua

```diff
@@ -0,0 +1,80 @@
+-------------------------------------------------------------------------------
+-- NOTE: the following is copied from lua-resty-mlcache:                     --
+------------------------------------------------------------------------ cut --
+local cjson = require "cjson.safe"
+
+
+local type = type
+local error = error
+local tostring = tostring
+local fmt = string.format
+local now = ngx.now
+
+
+local TYPES_LOOKUP = {
+  number  = 1,
+  boolean = 2,
+  string  = 3,
+  table   = 4,
+}
+
+
+local marshallers = {
+  shm_value = function(str_value, value_type, at, ttl)
+      return fmt("%d:%f:%f:%s", value_type, at, ttl, str_value)
+  end,
+
+  shm_nil = function(at, ttl)
+      return fmt("0:%f:%f:", at, ttl)
+  end,
```

Comment:
why not
```
[0] = function() return "" end
```
?

Reply:
Not sure, it is just a copy paste from here:
https://github.com/thibaultcha/lua-resty-mlcache/blob/master/lib/resty/mlcache.lua#L66-L96

Reply:
So that `nil` values can take a shortcut branch and avoid a bunch of table lookups.